modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/plugin/PluginClassLoaderConfigurationAssembler.java

```diff
@@ -0,0 +1,132 @@
+          .filter(bundleDescriptor -> bundleDescriptor.getDescriptor().getGroupId()
+              .equals(this.bundleDescriptor.getGroupId())
+              && bundleDescriptor.getDescriptor().getArtifactId().equals(this.bundleDescriptor.getArtifactId()))
+          .filter(bundleDependency -> bundleDependency.getAdditionalDependenciesList() != null
+              && !bundleDependency.getAdditionalDependenciesList().isEmpty())
+          .forEach(bundleDependency -> processPluginAdditionalDependenciesURIs(bundleDependency,
+                                                                               artifactClassLoaderConfigurationBuilder)
+                                                                                   .forEach(uri -> {
+                                                                                     final URL dependencyArtifactUrl;
+                                                                                     try {
+                                                                                       dependencyArtifactUrl = uri.toURL();
+                                                                                     } catch (MalformedURLException e) {
+                                                                                       throw new ArtifactActivationException(createStaticMessage(format("There was an exception obtaining the URL for the artifact [%s], file [%s]",
+                                                                                                                                                        artifactLocation
+                                                                                                                                                            .getAbsolutePath(),
+                                                                                                                                                        uri)),
+                                                                                                                             e);
+                                                                                     }
+                                                                                     dependenciesArtifactsUrls
+                                                                                         .add(dependencyArtifactUrl);
+                                                                                   }));
+    }
+
+    return dependenciesArtifactsUrls;
+  }
+
+  private List<URI> processPluginAdditionalDependenciesURIs(BundleDependency bundleDependency,
+                                                            ArtifactClassLoaderConfigurationBuilder artifactClassLoaderConfigurationBuilder) {
+    return bundleDependency.getAdditionalDependenciesList().stream().map(additionalDependency -> {
+      // TODO: check if the dependency belongs to the deny-list
```

Comment:
i saw this same TODO somewhere else before...

Reply:
This is for a plugin's additional dependencies, and the other one is for the plugin itself.